### PR TITLE
Adapter Network

### PR DESCRIPTION
**Pull request by yyycggg**
Re: https://github.com/cocos/3d-tasks/issues/13998

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by PPpro — APPROVED** [review]

**Comment by PPpro** [comment]
please comfirm to Taobao 
whether we can create multiple instances of WebSocket

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 1]
```diff
@@ -26,62 +31,82 @@ export default class WebSocket {
...
-      }
-    })
+      this._triggerEvent('open')
+    }
+    my.onSocketOpen(this._onopen)
 
-    socketTask.onError((res) => {
-      if (typeof this.onerror === 'function') {
-        this.onerror(new Error(res.errMsg))
+    this._onmessage = (res) => {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: Why rename to `_onmessage`? It combines two words, so i think `_onMessage` is better.

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 2]
```diff
@@ -26,62 +31,82 @@ export default class WebSocket {
...
+      if (res && res.data && res.isBuffer) {
+        res.data = _utils.base64ToArrayBuffer(res.data);
       }
-    })
+      this._triggerEvent('message', res)
+    }
+    my.onSocketMessage(this._onmessage)
+
+    this._onerror = (res) => {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: The same as above.

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js [hunk 3]
```diff
@@ -26,62 +31,82 @@ export default class WebSocket {
...
+      this._triggerEvent('error', res)
+    }
+    my.onSocketError(this._onerror)
+
+    this._onclose = () => {
```
Comment (minggo) [inline platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: The same as above.

File: platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js (hunk 1 above)
Comment (yyycggg) [reply platforms/minigame/platforms/taobao/wrapper/builtin/WebSocket.js]: OK

**Review by PPpro — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**